app/src/features/Preview/Preview.tsx

```diff
@@ -0,0 +1,262 @@
+    marginBottom: theme.spacing(2),
+    textTransform: "none",
+    color: theme.palette.text.secondary,
+    "&:hover": {
+      backgroundColor: "inherit",
+      color: theme.palette.text.primary,
+    },
+  },
+}));
+
+type Exploration = { fields: string[]; rows: string[][] };
+
+const Preview = (): JSX.Element => {
+  const classes = useStyles();
+  const { t } = useTranslation();
+  const history = useHistory();
+  const prefersDarkMode = useMediaQuery("(prefers-color-scheme: dark)");
+  const { mappingId } = useParams<{
+    mappingId?: string;
+  }>();
+
+  const [alert, setAlert] = useState<string | undefined>(undefined);
+  const handleAlertClose = () => setAlert(undefined);
+
+  const [preview, setPreview] = useState<Record<string, unknown> | undefined>(
+    undefined
+  );
+
+  const { data: mapping } = useApiResourcesRetrieveQuery(
+    { id: mappingId ?? "" },
+    { skip: mappingId === undefined }
+  );
+
+  const { data: owner } = useApiOwnersRetrieveQuery(
+    {
+      id: mapping?.primary_key_owner ?? "",
+    },
+    { skip: mapping === undefined }
+  );
+
+  const {
+    data: exploration,
+    isLoading: isExplorationLoading,
+  } = usePagaiExploreRetrieveQuery(
+    {
+      resourceId: mapping?.id ?? "",
+      owner: owner?.name ?? "",
+      table: mapping?.primary_key_table ?? "",
+    },
+    { skip: mapping === undefined || owner === undefined }
+  );
+
+  const [apiPreviewCreate] = useApiPreviewCreateMutation();
+
+  const handleFhirIconClick = (index: number) => async () => {
+    const primaryKey = mapping?.primary_key_table;
+    if (exploration && primaryKey && mapping) {
+      const indexPrimaryKey = (exploration as Exploration).fields.indexOf(
+        primaryKey
+      );
+      const primaryKeyValue = (exploration as Exploration).rows[index]?.[
+        indexPrimaryKey
+      ];
+      try {
+        const previewResult = await apiPreviewCreate({
+          previewRequest: {
+            resource_id: mapping?.id,
+            primary_key_values: [primaryKeyValue ?? ""],
+          },
+        });
+        setPreview(previewResult);
+      } catch (e) {
+        setAlert(e.message);
+      }
+    }
+  };
+
+  const handleCancelClick = () => {
+    history.goBack();
+  };
+
+  const FhirIconCell = () => (
+    <IconButton size="small" className={classes.icon}>
+      <Icon
+        icon={IconNames.FLAME}
+        iconSize={15}
+        className={classes.iconFlame}
+      />
+    </IconButton>
+  );
+
+  if (isExplorationLoading) return <CircularProgress />;
+  return (
+    <Container className={classes.container} maxWidth="xl">
+      <Button
+        className={classes.button}
+        startIcon={<ArrowBack />}
+        onClick={handleCancelClick}
+        disableRipple
+      >
+        <Typography>{t("back")}</Typography>
+      </Button>
+      <TableContainer className={classes.table}>
+        <Table size="small">
+          <TableHead>
+            <TableRow className={classes.cellsTitle}>
```

Comment:
I don't really like the className naming here. We're applying a class `cellsTitle` on an entire row